### ISSUE TITLE
GPU Not Detected on NVIDIA Jetson Orin Nano Super 8GB

### ISSUE THREAD

**Issue by davistroy** [Pavelevich/llm-checker#47]
davistroy@jetson:~$ llm-checker smart-recommend

 ███████╗███╗   ███╗ █████╗ ██████╗ ████████╗
 ██╔════╝████╗ ████║██╔══██╗██╔══██╗╚══██╔══╝
 ███████╗██╔████╔██║███████║██████╔╝   ██║
 ╚════██║██║╚██╔╝██║██╔══██║██╔══██╗   ██║
 ███████║██║ ╚═╝ ██║██║  ██║██║  ██║   ██║
 ╚══════╝╚═╝     ╚═╝╚═╝  ╚═╝╚═╝  ╚═╝   ╚═╝
          RECOMMEND - AI Powered

✔ Analysis complete!

=== Hardware Analysis ===
  Cortex-A78AE (7GB RAM, CPU-only)
  Tier: ULTRA LOW
  Backend: cpu
  Max model size: 3GB

=== Top Recommendations ===

[BEST] Best Overall:
  qwen2.5-coder:1.5b-base-q5_K_M
  1.5B params | 0.9GB | Q5_K_M
  Score: 65/100 (Q:90 S:10 F:100)
  ~4 tokens/sec
  ollama pull qwen2.5-coder:1.5b-base-q5_K_M

⚡ Fastest:
  granite3.1-moe:3b-instruct-q3_K_M
  3B | 1.2GB | ~7 tok/s
  ollama pull granite3.1-moe:3b-instruct-q3_K_M

Highest Quality:
  qwen2.5-coder:3b-base-q8_0
  3B | 3GB | Quality: 96/100
  ollama pull qwen2.5-coder:3b-base-q8_0

=== Other Good Options ===
[65] qwen2.5-coder:1.5b-instruct-q5_K_M - 1.5B, 0.9GB
[65] qwen2.5:1.5b-instruct-q8_0 - 1.5B, 1.5GB
[65] qwen2.5-coder:1.5b-base-q8_0 - 1.5B, 1.5GB
[65] qwen2.5-coder:1.5b-instruct-q8_0 - 1.5B, 1.5GB

**Comment by Pavelevich** [comment]
Thanks for the detailed repro — this is now prioritized and I’ve reopened the issue until we confirm Jetson detection is fixed on your device.

Two key notes from your output:
1. You’re still on `3.2.8`; the Jetson fallback detection patch landed after that build.
2. `--models` is not a valid flag for `smart-recommend` (that flag is supported in `ai-check`).

Please update to latest `main` and re-run:

```bash
llm-checker --version
llm-checker hw-detect
llm-checker recommend --no-verbose
llm-checker ai-check --models llama3.2:1b phi3.5:latest mistral:7b
```

If GPU still shows as not detected, paste those outputs plus:

```bash
command -v nvidia-smi || echo "nvidia-smi missing"
cat /etc/nv_tegra_release 2>/dev/null || echo "nv_tegra_release missing"
```

I’ll patch quickly based on that.

**Comment by Pavelevich** [comment]
Thanks — this report identifies the package source mismatch.\n\nYou installed **@pavelevich/llm-checker**, but the actively released CLI is **llm-checker** (unscoped).\nThat is why your binary still reports **3.2.8** while current latest is newer.\n\nPlease run exactly:\n\n```bash\nnpm uninstall -g @pavelevich/llm-checker\nnpm install -g llm-checker@latest\nwhich llm-checker\nllm-checker --version\n```\n\nIf version still shows 3.2.8, clear shell path cache and re-check:\n\n```bash\nhash -r\nllm-checker --version\n```\n\nThen rerun:\n\n```bash\nllm-checker hw-detect\nllm-checker recommend --no-verbose\n```\n\nIf GPU is still not detected after switching to unscoped latest, share the new outputs and I’ll patch immediately.

**Comment by davistroy** [comment]
davistroy@jetson:~$ npm uninstall -g @pavelevich/llm-checker

removed 151 packages in 769ms
davistroy@jetson:~$ npm install -g llm-checker@latest

changed 151 packages in 9s

46 packages are looking for funding
  run `npm fund` for details
davistroy@jetson:~$ which llm-checker
/home/davistroy/.nvm/versions/node/v24.13.1/bin/llm-checker
davistroy@jetson:~$ llm-checker --version
3.4.0
davistroy@jetson:~$ llm-checker hw-detect

 ██╗  ██╗ █████╗ ██████╗ ██████╗ ██╗    ██╗ █████╗ ██████╗ ███████╗
 ██║  ██║██╔══██╗██╔══██╗██╔══██╗██║    ██║██╔══██╗██╔══██╗██╔════╝
 ███████║███████║██████╔╝██║  ██║██║ █╗ ██║███████║██████╔╝█████╗
 ██╔══██║██╔══██║██╔══██╗██║  ██║██║███╗██║██╔══██║██╔══██╗██╔══╝
 ██║  ██║██║  ██║██║  ██║██████╔╝╚███╔███╔╝██║  ██║██║  ██║███████╗
 ╚═╝  ╚═╝╚═╝  ╚═╝╚═╝  ╚═╝╚═════╝  ╚══╝╚══╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝
                      DETECTION

✔ Hardware detected!

=== Hardware Detection ===

Summary:
  NVIDIA Jetson Orin Nano (6GB VRAM) + Cortex-A78AE
  Tier: LOW
  Max model size: 4GB
  Best backend: cuda

CPU:
  Cortex-A78AE
  Cores: 6 (6 physical)
  SIMD: none

CUDA:
  Driver: null
  CUDA: 12.6
  Total VRAM: 6GB
  NVIDIA Jetson Orin Nano: 6GB

Fingerprint: cuda--jetson-orin-nano-6gb

davistroy@jetson:~$
